pathway-analysis/qusage_replicate_vignette.Rmd

```diff
@@ -0,0 +1,404 @@
+par(mar = c(13, 4, 1, 2))
+plotCIs(timepoint_qusage_results)
+dev.off()
+```
+
+Let's look more closely at the T Cell Receptor Signaling Pathway. 
+We can see above that this pathway has a negative fold change and therefore
+should be higher at `t0`.
+
+```{r}
+pathway_index <- 
+  which(names(kegg_genesets) == "KEGG_T_CELL_RECEPTOR_SIGNALING_PATHWAY")
+```
+
+Plot the overall distribution of the pathway (thick black line) and the 
+distribution of the individual genes (thin lines colored by their standard
+deviations).
+
+```{r}
+plotGeneSetDistributions(timepoint_qusage_results, path.index = pathway_index)
+```
+
+We can see the majority of genes in this pathway don't change much, but a 
+handful have lower values in `t1`.
+
+Now, plot the mean and confidence intervals for each gene in the pathway.
+This gives us insight into _which_ genes changed the most.
+
+```{r}
+plotCIsGenes(timepoint_qusage_results, path.index = pathway_index)
+```
+
+We can easily check to see if the directionality is as we expect by making
+a boxplot of the gene `387` which has a negative value and is on right side of 
+the graph.
+
+```{r}
+graphics::boxplot(exprs_mat["387", ] ~ filtered_metadata_df$timepoint)
+```
+
+### Two way comparison
+
+Patients in this influenza data set were either _symptomatic_ or _asymptomatic_.
+We can check if there are differences in pathway _responses_ to influenza virus
+(e.g., between time points) between the two groups.
+The ability to do this more complex comparison is a great feature of `qusage`.
+Again, we'll mirror the analysis performed in the vignette.
+
+The symptomatic or asymptomatic information is in the `clinic_pheno` column
+of the metadata.
+
+```{r}
+head(filtered_metadata_df$clinic_pheno)
+```
+
+Generate labels for the two way comparison.
+
+```{r}
+twoway_labels <- paste(filtered_metadata_df$clinic_pheno, 
+                       filtered_metadata_df$timepoint,
+                       sep = ".")
+twoway_labels
+```
+
+Symptomatic comparison
+
+```{r}
+sx_results <- qusage(eset = exprs_mat,
+                     labels = twoway_labels,
+                     contrast = "Symptomatic.t1-Symptomatic.t0",
+                     # timepoints are paired samples, so we can specify that
+                     # information here
+                     geneSets = kegg_genesets,
+                     pairVector = filtered_metadata_df$subject)
+```
+
+Asymptomatic comparison
+
+```{r}
+asx_results <- qusage(eset = exprs_mat,
+                      labels = twoway_labels,
+                      contrast = "Asymptomatic.t1-Asymptomatic.t0",
+                      # timepoints are paired samples, so we can specify that
+                      # information here
+                      geneSets = kegg_genesets,
+                      pairVector = filtered_metadata_df$subject)
+```
+
+Calculate the difference between the two comparisons
+
+```{r}
+asx_vs_sx <- qusage(eset = exprs_mat, 
```

Comment:
`asx_vs_sx ` or `sx_vs_asx` Is there an order that makes more sense? In the contrast you have `Symptomatic` first. Should it parallel in the name or nah?